server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyLDAPUsersDAO.java

```diff
@@ -173,8 +173,8 @@ void dispose() {
         ldapConnectionPool.close();
     }
 
-    private Filter createFilter(String username) {
-        Filter specificUserFilter = Filter.createEqualityFilter(ldapConfiguration.getUserIdAttribute(), username);
+    private Filter createFilter(String username, String ldapRetrievalUserAttribute) {
```

Comment:
`ldapRetrievalUserAttribute` supposed to be Optional no?

Reply:
Yes, otherwise it should fallback to `UserIdAttribute`